src/components/ContactsList/_modals/ContactUpsertModal/ContactUpsertModal.tsx

```diff
@@ -0,0 +1,120 @@
+import { Modal } from '@/components/Modal';
+import { FC, useState } from 'react';
+import { SrOnlySpan } from '@/styles';
+import { Loader } from '@/components/Loader';
+import { PaymailInput } from '@/components/Input/PaymailInput';
+import { Input } from '@/components/Input';
+import { upsertContact } from '@/api/requests/contact';
+import { modalCloseTimeout } from '@/components/Modal/modalCloseTimeout';
+import { ErrorBar } from '@/components/ErrorBar';
+import { isValidPhone } from '@/utils/helpers/validatePhone';
+import { ContactFields } from './useContactFields';
+import { EMAIL_REGEX } from '@/utils/constants';
+
+type ContactUpsertModal = {
+  onSubmitted: () => void;
+  onCancel: () => void;
+  sucessMsg: string;
+  errorMsg: string;
+  modalTitle: string;
+  modalSubtitle: string;
+  fields: ContactFields;
+  disabledPaymailInput: boolean;
+};
+
+export const ContactUpsertModal: FC<ContactUpsertModal> = ({
+  onSubmitted,
+  onCancel,
+  fields,
+  errorMsg,
+  sucessMsg,
+  modalTitle,
+  modalSubtitle,
+  disabledPaymailInput,
+}) => {
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState<string | null>(null);
+  const [successMessage, setSuccessMessage] = useState<string>('');
+  const { name, paymail, phone, setName, setPaymail, setPhone } = fields;
```

Comment:
Add useEffect from `ContactEditModal` and call `useContactFields` here

Reply:
Can't do that. See above comment 👆